mtp_transaction_uploader/upload.py

```diff
@@ -52,6 +53,12 @@ def retrieve_data_services_files():
                 m = file_pattern.match(filename)
 
                 if m:
+                    stat = conn.stat(filename)
+                    if stat.st_size > SIZE_LIMIT_BYTES:
+                        print("%s is too large (%s), download skipped."
+                              % (filename, stat.st_size))
+                        continue
```

Comment:
I'd really like to see this tested… Ideally, we should have `SIZE_LIMIT_BYTES` be customisable so we don't have to create a 50MB test file, though?